atlassian_jwt_auth/signer.py

```diff
@@ -50,7 +64,7 @@ def generate_jwt(self, audience, **kwargs):
             self.issuer)
         return jwt.encode(
             self._generate_claims(audience, **kwargs),
-            key=private_key_pem,
+            key=self._private_key,
```

Comment:
This almost halves the time taken to encode a RS256 jwt (after the first access to `_private_key`).

Reply:
Hmmm... I'm not getting the same results when I profile this function (master and your branch are essentially the same). I don't think the `@property` decorator does any sort of caching or memoization of the call; it is just syntatic sugar so you don't have to call the method with parenthesis.

I created a branch where we explicitly check for `self._private_key is None` and then populate it if it is not present. The benchmarks are about 50% lower with that change. I think that may have been what you were shooting for?

https://github.com/atlassian/asap-authentication-python/pull/92